int32 id
string name
bool is_target
KeyValue[] attr
KeyValue[] exclude_attr
